### ISSUE TITLE
[BUG] process-agent concurrent map iteration and map write

### ISSUE THREAD

**Issue by jcogilvie** [DataDog/datadog-agent#19034]
**Agent Environment**
Kubernetes, agent image `agent:7.46.0`

**Describe what happened:**
Fatal error in `process-agent`:

[gist](https://gist.github.com/jcogilvie/820cb953116768842f4b907ecaea1f1e)

**Describe what you expected:**
Continued successful operation

**Steps to reproduce the issue:**
Kubernetes container config:

```
    - name: process-agent
      image: public.ecr.aws/datadog/agent:7.46.0
      command:
        - process-agent
        - '--cfgpath=/etc/datadog-agent/datadog.yaml'
      env:
        - name: GODEBUG
          value: x509ignoreCN=0
        - name: DD_API_KEY
          valueFrom:
            secretKeyRef:
              name: datadog
              key: api-key
        - name: DD_REMOTE_CONFIGURATION_ENABLED
          value: 'false'
        - name: DD_AUTH_TOKEN_FILE_PATH
          value: /etc/datadog-agent/auth/token
        - name: DD_CLUSTER_NAME
          value: REDACTED
        - name: DD_TAGS
          value: EKS:REDACTED
        - name: KUBERNETES
          value: 'yes'
        - name: DD_KUBERNETES_KUBELET_HOST
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.hostIP
        - name: DD_HISTOGRAM_PERCENTILES
          value: 0.90 0.95 0.99
        - name: DD_LOG_FORMAT_JSON
          value: 'true'
        - name: DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_GRPC_ENDPOINT
          value: 0.0.0.0:4317
        - name: DD_OTLP_CONFIG_RECEIVER_PROTOCOLS_HTTP_ENDPOINT
          value: 0.0.0.0:4318
        - name: DD_CLUSTER_AGENT_ENABLED
          value: 'true'
        - name: DD_CLUSTER_AGENT_KUBERNETES_SERVICE_NAME
          value: datadog-cluster-agent
        - name: DD_CLUSTER_AGENT_AUTH_TOKEN
          valueFrom:
            secretKeyRef:
              name: datadog-cluster-agent
              key: token
        - name: DD_PROCESS_AGENT_ENABLED
          value: 'true'
        - name: DD_PROCESS_AGENT_DISCOVERY_ENABLED
          value: 'true'
        - name: DD_LOG_LEVEL
          value: INFO
        - name: DD_SYSTEM_PROBE_ENABLED
          value: 'true'
        - name: DD_SYSTEM_PROBE_NETWORK_ENABLED
          value: 'true'
        - name: DD_DOGSTATSD_SOCKET
          value: /var/run/datadog/statsd.sock
        - name: DD_ORCHESTRATOR_EXPLORER_ENABLED
          value: 'true'
      resources:
        limits:
          cpu: 100m
          memory: 512Mi
        requests:
          cpu: 60m
          memory: 200Mi
      volumeMounts:
        - name: config
          readOnly: true
          mountPath: /etc/datadog-agent
        - name: logdatadog
          mountPath: /var/log/datadog
        - name: auth-token
          readOnly: true
          mountPath: /etc/datadog-agent/auth
        - name: dsdsocket
          mountPath: /var/run/datadog
        - name: tmpdir
          mountPath: /tmp
        - name: os-release-file
          readOnly: true
          mountPath: /host/etc/os-release
        - name: runtimesocketdir
          readOnly: true
          mountPath: /host/var/run
          mountPropagation: None
        - name: cgroups
          readOnly: true
          mountPath: /host/sys/fs/cgroup
          mountPropagation: None
        - name: passwd
          readOnly: true
          mountPath: /etc/passwd
        - name: procdir
          readOnly: true
          mountPath: /host/proc
          mountPropagation: None
        - name: sysprobe-socket-dir
          readOnly: true
          mountPath: /var/run/sysprobe
        - name: sysprobe-config
          readOnly: true
          mountPath: /etc/datadog-agent/system-probe.yaml
          subPath: system-probe.yaml
        - name: kubeletlogs
          mountPath: /var/log/journal
        - name: kube-api-access-kwl4z
          readOnly: true
          mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      terminationMessagePath: /dev/termination-log
      terminationMessagePolicy: File
      imagePullPolicy: IfNotPresent
```

**Additional environment details (Operating System, Cloud provider, etc):**
AWS EKS 1.26

**Comment by robertjli** [comment]
Hello,

We believe this has been addressed with https://github.com/DataDog/datadog-agent/pull/16803, so upgrading to 7.47 should resolve the issue.

**Comment by davidor** [comment]
Closing as this should be fixed in 7.47 as mentioned above.
